src/pages/workspace/WorkspaceInitialPage.tsx

```diff
@@ -449,7 +449,13 @@ function WorkspaceInitialPage({policyDraft, policy: policyProp, route}: Workspac
             >
                 <HeaderWithBackButton
                     title={policyName}
-                    onBackButtonPress={() => Navigation.goBack(ROUTES.SETTINGS_WORKSPACES)}
+                    onBackButtonPress={() => {
+                        if (route.params?.backTo) {
+                            Navigation.goBack(route.params?.backTo);
+                            return;
+                        }
+                        Navigation.goBack(ROUTES.SETTINGS_WORKSPACES);
```

Comment:
Let's please change this to `Navigation.goBack(route.params?.backTo ?? ROUTES.SETTINGS_WORKSPACES);`